core/src/retransmit_stage.rs

```diff
@@ -122,6 +126,36 @@ impl RetransmitStats {
                 stats.num_shreds_skipped.into_inner(),
                 i64
             ),
+            (
+                "num_shreds_received_root",
+                stats.num_shreds_received[0].load(Ordering::Relaxed),
+                i64
+            ),
+            (
+                "num_shreds_received_1st_layer",
+                stats.num_shreds_received[1].load(Ordering::Relaxed),
+                i64
+            ),
+            (
+                "num_shreds_received_2nd_layer",
+                stats.num_shreds_received[2].load(Ordering::Relaxed),
+                i64
+            ),
+            (
+                "num_shreds_sent_root",
+                stats.num_shreds_sent[0].load(Ordering::Relaxed),
+                i64
+            ),
+            (
+                "num_shreds_sent_1st_layer",
+                stats.num_shreds_sent[1].load(Ordering::Relaxed),
+                i64
+            ),
+            (
+                "num_shreds_sent_2nd_layer",
+                stats.num_shreds_sent[2].load(Ordering::Relaxed),
+                i64
+            ),
```

Comment:
This is pretty cool!

Do you think it would be useful to organize these metrics per slot in the blockstore metrics and report them upon block completion?

I'm imagining a test where we grab a staked validator on testnet, and then push some shreds for one of it's leader paths where:
1.  We know all the paths
2.  We collect this analysis on a per-slot basis

And hopefully that will allow us to drill down into where shreds are getting lost/stuck/turbine tree is inconsistent

Reply:
+1 - Reporting by slot would be hugely beneficial

Reply:
done